packages/common/core-interfaces/package.json

```diff
@@ -87,11 +100,37 @@
 		"lint": "fluid-build . --task lint",
 		"lint:fix": "fluid-build . --task eslint:fix --task format",
 		"place:cjs:package-stub": "copyfiles -f ../../../common/build/build-common/src/cjs/package.json ./dist",
+		"test": "npm run test:mocha",
+		"test:coverage": "c8 npm test",
+		"test:mocha": "npm run test:mocha:esm && echo skipping cjs to avoid overhead - npm run test:mocha:cjs",
+		"test:mocha:cjs": "mocha --recursive \"dist/test/**/*.spec.*js\" --exit",
+		"test:mocha:esm": "mocha --recursive \"lib/test/**/*.spec.*js\" --exit",
+		"test:mocha:verbose": "cross-env FLUID_TEST_VERBOSE=1 npm run test:mocha",
 		"tsc": "fluid-tsc commonjs --project ./tsconfig.cjs.json && npm run place:cjs:package-stub",
 		"tsc:watch": "npm run place:cjs:package-stub && fluid-tsc commonjs --project ./tsconfig.cjs.json --watch",
 		"typetests:gen": "flub generate typetests --dir . -v",
 		"typetests:prepare": "flub typetests --dir . --reset --previous --normalize"
 	},
+	"c8": {
```

Comment:
There are a lot of changes in this PR that don't seem related or explained in the PR description. Perhaps you could front load those where possible (ex: add support for beta APIs as a change, and add support for testing in this package as another) so this change could focus on the types. I don't think I'm a great reviewer for all this package setup, script, api extractor and linter changes (at least I can't confidently say they are right, and have a reason to be made as part of this change)

Reply:
All are PR related.
- The API extractor changes were for adding `@beta` support, which should only be added when there is a `@beta` API. Changes are reduced to `@legacy` and support changes are now removed.
- The test changes are for adding test support. Previously there were no tests here. Can't add test support without having tests. Now noted in PR description.